h/static/scripts/directive/annotation.coffee

```diff
@@ -202,6 +208,10 @@ AnnotationController = [
       switch @action
         when 'create'
           updateDomainModel(model, @annotation)
+
+          if group.focusedGroup().hashid != '__public__'
```

Comment:
> Should the client even know these are hashids? Shouldn't they just be opaque identifiers here, and the hash bit is an implementation detail hidden within the server?

At least two bits of the client code need to know about hashids:

1. When the client receives an annotation from the server it contains `group: "<hashid>"` and the client needs to take that hashid and transform it into the full group dict (including name).

2. When the client saves an annotation it has to send `group: "<hashid>"` to the server, so it needs to pull just the hashid part out of whatever representation of the group it's using.

I could have a look and see if the rest of the code could return and pass full `{name: "<name>", hashid: "<hashid>"} group objects around, and most of the code could just work with `.name` and never know about `.hashid` (but blindly pass the whole object around to the hashid does stay with it).